vega_sim/scenario/common/agents.py

```diff
@@ -709,3 +710,151 @@ def finalise(self):
+            spread (int): Spread of the current markets market maker.
+            initial_asset_mint (float, optional): Quantity of the asset the agent should initally mint.
+            buy_intensity (float, optional): Median of the poission distribution used to sample the volume of buy orders.
+            sell_intensity (float, optional): Median of the poission distribution used to sample the volume of sell orders.
+            tag (str, optional): String to tag to the market and asset name.
+            random_state: (np.random.RandomState, optional): Object for creating distributions to randomly sampling from.
+        """
+
+        super().__init__(wallet_name + str(tag), wallet_pass)
+
+        self.market_name = market_name
+        self.asset_name = asset_name
+        self.spread = spread
+        self.initial_asset_mint = initial_asset_mint
+        self.buy_intensity = buy_intensity
+        self.sell_intensity = sell_intensity
+        self.tag = tag
+        self.random_state = (
+            random_state if random_state is not None else np.random.RandomState()
+        )
+
+    def initialise(self, vega: VegaServiceNull):
+        """Initialise the agents wallet and mint the required market asset.
+
+        Args:
+            vega (VegaServiceNull): Object running a locally-hosted Vega service.
+        """
+
+        super().initialise(vega=vega)
+
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        self.vega.mint(
+            self.wallet_name,
+            asset=self.asset_id,
+            amount=self.initial_asset_mint,
+        )
+        self.vega.wait_fn(2)
+
+    def step(self, vega_state: VegaState):
+        """Random submits and cancels limit orders.
+
+        Args:
+            vega_state (VegaState): Object describing the state of the network and the state of the market.
+        """
+
+        if (
+            vega_state.market_state[self.market_id].trading_mode
+            == markets_protos.Market.TradingMode.TRADING_MODE_CONTINUOUS
+        ) and vega_state.market_state[
+            self.market_id
+        ].state == markets_protos.Market.State.STATE_ACTIVE:
+
+            agent_submit_order = self.random_state.choice([0, 1, 1])
+            if agent_submit_order:
+                self._sumbit_order()
+
+            agent_cancel_order = self.random_state.choice([0, 0, 1])
+            if agent_cancel_order:
+                self._cancel_order(vega_state=vega_state)
+
+    def _sumbit_order(self):
+
+        best_bid_price, best_offer_price = self.vega.best_prices(
+            market_id=self.market_id
+        )
+
+        agent_order_side = self.random_state.choice(["buy", "sell"])
+
+        if agent_order_side == "buy":
+
+            volume = self.random_state.poisson(self.buy_intensity)
```

Comment:
Can we update this to multiply by a passed in "trade unit size"? I recently realised (but haven't yet updated prod to fix) that just directly selecting from poisson means we can only do unit order sizes.

Reply:
Sure, have replaced buy and sell intensity arguments with buy and sell volume arguments

Reply:
Ah, may have been unclear, keep the poisson randomness but multiply buy the passed buy/sell volume (so we want both parameters :) )